editor/test/editor/defold_project_search_test.clj

```diff
@@ -8,69 +8,77 @@
             [integration.test-util :as test-util]
             [support.test-support :refer [with-clean-system]]
             [util.thread-util :as thread-util])
-  (:import [java.util.concurrent LinkedBlockingQueue]))
+  (:import [java.util.concurrent LinkedBlockingQueue TimeUnit]))
 
 (def ^:const search-project-path "test/resources/search_project")
-(def ^:const timeout-ms 15000)
+(def ^:const timeout-ms 1000)
 
 (defn- make-consumer [report-error!]
-  (atom {:consumed [] :future nil :report-error! report-error!}))
-
-(defn- consumer-start! [consumer-atom results-fn!]
-  (swap! consumer-atom
-         (fn [consumer]
-           (when (:future consumer)
-             (future-cancel (:future consumer)))
```

Comment:
Woah, that was really dangerous — side-effects in a retriable block!

Reply:
It actually wasn't — `future-cancel` is idempotent.

Reply:
Yep. That tripped me up as well! 🙂